mmdet/models/losses/ae_loss.py

```diff
@@ -0,0 +1,71 @@
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from ..builder import LOSSES
+
+
+def ae_loss_per_image(tl_preds, br_preds, match):
+    tl_list, br_list, me_list = [], [], []
+    for m in match:
+        [tl_y, tl_x], [br_y, br_x] = m
+        tl_e = tl_preds[:, tl_y, tl_x].view(-1, 1)
+        br_e = br_preds[:, br_y, br_x].view(-1, 1)
+        tl_list.append(tl_e)
+        br_list.append(br_e)
+        me_list.append((tl_e + br_e) / 2.0)
+
+    tl_list = torch.cat(tl_list)
+    br_list = torch.cat(br_list)
+    me_list = torch.cat(me_list)
+
+    # assert tl_list.size() == br_list.size()
+
+    N, M = tl_list.size()
+
+    if N > 0:
+        pull_loss = (tl_list - me_list).pow(2) + (br_list - me_list).pow(2)
+        pull_loss = pull_loss.sum() / N
+    else:
+        pull_loss = torch.tensor(0).type_as(me_list).requires_grad_()
```

Comment:
Please refer to https://github.com/open-mmlab/mmdetection/blob/8951a8a70f36d442d8f5a9ecfe6361bbd6dc2adb/mmdet/models/losses/pisa_loss.py#L149 for 0 loss.